test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/DotnetCommand.cs

```diff
@@ -15,6 +17,41 @@ public DotnetCommand()
         public DotnetCommand(string dotnetUnderTest)
             : base(dotnetUnderTest)
         {
+            // Some scripts (e.g. RunCsc/RunVbc) depend on the correct `dotnet`
+            // being in the PATH. When tests use `DotnetUnderTest.WithBackwardsCompatibleRuntimes`,
+            // this means that the up-to-date dotnet command is resolved by the
+            // old-runtime-targeting RunCsc/Roslyn, causing compatability issues.
+            // So, detect if we're running a dotnet that's not just a plain "dotnet",
+            // and if so, make sure it's on the PATH.
+            var containingDirectory = Path.GetDirectoryName(dotnetUnderTest);
+            if (!string.IsNullOrEmpty(containingDirectory))
+            {
+                // if it's already set, use that
+                string path;
+                if (!Environment.TryGetValue("PATH", out path))
+                {
+                    // otherwise grab it from the environment
```

Comment:
> If it is this case, could you add test coverage for this?

As you pointed out, this is part of tests, not shipped code - as far as I know, there isn't a good framework in place for thoroughly testing tests.

> will it fail the build upfront if I make a mistake in the code?

Yes, and this is why it was originally included - omissions or changes will fail the test runs. However, the purpose of comments is to make sure that future people who either read this code or would like to update this code understand the purpose of this code, and so know how to add additional tests for any functionality they may add or change.

Please do also note the updated commit - could I ask for clarification if your above comments were for the commit this comment is on, or the most recent commit?